src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3578,35 +3600,20 @@ static void set_coredump_filter(CoredumpFilterBit bit) {
 
 static size_t _large_page_size = 0;
 
-size_t os::Linux::find_default_large_page_size() {
-  if (_default_large_page_size != 0) {
-    return _default_large_page_size;
-  }
-  size_t large_page_size = 0;
+static size_t scan_default_large_page_size() {
+  size_t default_large_page_size = 0;
 
   // large_page_size on Linux is used to round up heap size. x86 uses either
   // 2M or 4M page, depending on whether PAE (Physical Address Extensions)
   // mode is enabled. AMD64/EM64T uses 2M page in 64bit mode. IA64 can use
-  // page as large as 256M.
+  // page as large as 1G.
   //
   // Here we try to figure out page size by parsing /proc/meminfo and looking
   // for a line with the following format:
   //    Hugepagesize:     2048 kB
   //
   // If we can't determine the value (e.g. /proc is not mounted, or the text
-  // format has been changed), we'll use the largest page size supported by
-  // the processor.
-
-#ifndef ZERO
-  large_page_size =
-    AARCH64_ONLY(2 * M)
-    AMD64_ONLY(2 * M)
-    ARM32_ONLY(2 * M)
-    IA32_ONLY(4 * M)
-    IA64_ONLY(256 * M)
-    PPC_ONLY(4 * M)
-    S390_ONLY(1 * M);
-#endif // ZERO
+  // format has been changed), we'll set largest page size to 0
```

Comment:
I have a suspicion that this section change is causing the issue @tstuefe is seeing on linux ppc64. I think we are setting `size_t default_large_page_size = 0;` and then attempting an unfruitful scan of /proc/meminfo.

Later we get `alignment` using `lcm` in 
https://github.com/openjdk/jdk/blob/16ca370f1ac933a6aef49bd147d985e66b4c8930/src/hotspot/share/gc/shared/gcArguments.cpp#L81

https://github.com/openjdk/jdk/blob/af4cd04c2e393f8d1ffef60f49e3269adee649b8/src/hotspot/share/utilities/globalDefinitions.cpp#L370 

```
// least common multiple
size_t lcm(size_t a, size_t b) {
    size_t cur, div, next;

    cur = MAX2(a, b);
    div = MIN2(a, b);

    assert(div != 0, "lcm requires positive arguments");
```


But `os::large_page_size()` returns 0 instead of a default for the processor as before due to the new version of `scan_default_large_page_size()`.

This is my hunch from looking at the code. Still investigating and I will try setting a 64k page size in `os::vm_page_size()`.